mypy/checkexpr.py

```diff
@@ -2960,6 +2960,8 @@ def visit_index_with_type(self, left_type: Type, e: IndexExpr,
         elif (isinstance(left_type, CallableType)
               and left_type.is_type_obj() and left_type.type_object().is_enum):
             return self.visit_enum_index_expr(left_type.type_object(), e.index, e)
+        elif isinstance(left_type, TypeVarType):
+            return self.visit_index_with_type(left_type.upper_bound, e, original_type)
```

Comment:
What if this type var has `.values` instead of `.upper_bound`?

Reply:
surprisingly it works. I'm not sure why, but I suspect it's related to `TypeVar`s expansion in `checker.py::check_func_def`